webpack/assets/javascripts/react_app/routes/Settings/SettingsPage/components/SettingForm/SettingValueField.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Col, HelpBlock, FormGroup } from 'patternfly-react';
+
+import SettingValueArraySelect from './SettingValueArraySelect';
+import SettingValueHashSelect from './SettingValueHashSelect';
+import SettingValueBoolean from './SettingValueBoolean';
+import InputField from '../../../../../components/common/forms/InputField';
+
+const SettingValueField = ({ setting, form, field }) => {
+  const { selectValues } = setting;
+  let inputField = <InputField field={field} />;
+
+  const error = form.errors && form.errors.value;
+
+  if (selectValues && selectValues.kind === 'array') {
```

Comment:
This seems very `if`-based could we somehow leverage the InputFactory, just by adding types in it?
It seems ready for the job we are trying to achieve here IIRC :thinking: